### PR TITLE
WIP:Add a code to generate camera definition file

### PR DESCRIPTION
**Pull request by SaliniVenate**
A code to generate camera definition file by querying the camera device is added to test folder. Code has dependency on tinyxml which can be installed using:
$ sudo apt install libtinyxml-dev

To run the code:
$ cd camera-streaming-daemon/test
$ make
$./export_to_xml

### CODE REVIEW

**Comment by lbegani** [comment]
@SaliniVenate 
1. The name of the file should be changed to indicate that the code is to generate camera definition file for v4l2 devices. Eg:  export_v4l2_param_xml or better
2. There should be an input argument for the video device. Eg: 0,1,2, etc
3. Src and make file should be in their own folder. Where is the makefile for the app?

**Comment by lbegani** [comment]
@hamishwillee This is an app to automatically generate camera definition xml file for v4l2 devices. It will be basic and the user might still have to manually edit the file.

File: test/export_to_xml.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+case 12 :parameter->SetAttribute("type","uint32");break;
+}
+}
+
+
+int main()
+{
+
+std::string paramName;
+int fd=open("/dev/video0",O_RDWR);
```
[the remark this answers is not in the document]
Comment (anselmolsm) [reply test/export_to_xml.cpp]: Can you guarantee it's always /dev/video0 ?

File: test/export_to_xml.cpp [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+}
+    query_ext_ctrl.id |= V4L2_CTRL_FLAG_NEXT_CTRL | V4L2_CTRL_FLAG_NEXT_COMPOUND;
+}
+if (errno != EINVAL) {
+    perror("VIDIOC_QUERY_EXT_CTRL");
+    exit(EXIT_FAILURE);
+}
+       
+doc.LinkEndChild( root );
+doc.SaveFile( "camera-def-file.xml" );
```
Comment (anselmolsm) [inline test/export_to_xml.cpp]: hardcoded output file name.

Comment (anselmolsm) [reply test/export_to_xml.cpp]: Check result of SaveFile

File: test/export_to_xml.cpp [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+#include<errno.h>
+#include<linux/videodev2.h>
+#include <sys/ioctl.h>
+#include <tinyxml.h>
+
+struct v4l2_querymenu querymenu;
+struct v4l2_capability cap;
+struct v4l2_query_ext_ctrl query_ext_ctrl;
+
+static void enumerate_menu (int fd,TiXmlElement * parameter)
```
Comment (hamishwillee) [inline test/export_to_xml.cpp]: fd also not a good name

File: test/export_to_xml.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply test/export_to_xml.cpp]: This is a template that will require modification - so no, won't always be /dev/video0. That said, should this be a command line option?

File: test/export_to_xml.cpp (hunk 2 above)
Comment (hamishwillee) [reply test/export_to_xml.cpp]: That should be a default and definitely a command line option

**Comment by hamishwillee** [comment]
Yes, adding to /tools directory a good idea. I think this should have command line options for setting file name and possibly for the device name (with defaults). Should self document on command line with -h.

**Comment by SaliniVenate** [comment]
Raised a new PR #147 with changes as per review comments.So closing this one.

**Pull request closed without merging**